### PR TITLE
aalhommada-database-week1

### PR DESCRIPTION
**Pull request by aalhommada**

### CODE REVIEW

File: Week1/node.js [hunk 1]
```diff
@@ -0,0 +1,184 @@
...
+  user: 'hyfuser',
+  password: 'hyfpassword',
+  database: 'world',
+});
+
+database.connect();
+
+// create database world
+
+database.query(`CREATE DATABASE IF NOT EXISTS world`, (error, result) => {
```
Comment (gajduk) [inline Week1/node.js]: if not exists 👍

File: Week1/node.js [hunk 2]
```diff
@@ -0,0 +1,184 @@
...
+// use world database
+database.query(`USE world`, (error, result) => {
+  if (error) throw error;
+  console.log('change to WORLD database');
+});
+
+//========== create tables : country , city , and insert data in it =================
+
+const table_of_country =
+  'CREATE TABLE IF NOT EXISTS country (name VARCHAR(50), continent VARCHAR(50), region VARCHAR(50), surface_area FLOAT, indep_year INT, population INT, life_expectancy, GNP FLOAT, GNPold FLOAT, local_name VARCHAR(50), government_form VARCHAR(50), head_of_state VARCHAR(50),capital VARCHAR(50))';
```
Comment (gajduk) [inline Week1/node.js]: what is the type of `life_expectancy` ?

File: Week1/node.js [hunk 3]
```diff
@@ -0,0 +1,184 @@
...
+var table_of_city =
+  'CREATE TABLE IF NOT EXISTS city (id INT, name VARCHAR(50),country_code VARCHAR(10), district VARCHAR(50), population INT)';
+
+database.query(table_of_city, (error, results) => {
+  if (error) throw error;
+  console.log('City table is created');
+});
+
+var insert_countries = [
+  "INSERT INTO country (name, continent,surface_area, population) VALUES ('Angola','Africa',1246700,12878000)",
```
Comment (gajduk) [inline Week1/node.js]: did you know that you can have multiple values in a single query
``` 
"INSERT INTO country (name,     continent,  surface_area,  population) VALUES "+
"                    ('Angola', 'Africa',   1246700,       12878000),   "+
"                    ('Syria',  'Asia',     185180,        18269868),   "+ // and so forth
```

File: Week1/node.js [hunk 4]
```diff
@@ -0,0 +1,184 @@
...
+  if (error) throw error;
+
+  console.log(`\n Countries with population more than 8 million:`);
+  result.forEach(country => {
+    console.log(`${country.Name} , ${country.Population}`);
+  });
+});
+
+// Names of the countries that have “land” in their names ?
+database.query(`SELECT * FROM country WHERE Name LIKE '%land%' `, (error, result) => {
```
Comment (gajduk) [inline Week1/node.js]: Using `select *` is not recommended in real systems. Be specific with the result and only return the columns that you need, in this case the name.

File: Week1/node.js [hunk 5]
```diff
@@ -0,0 +1,184 @@
...
+
+  console.log(`\n Countries in Europe continent:`);
+  result.forEach(country => {
+    console.log(country.Name);
+  });
+});
+
+// Countries in descending order, based on their surface areas.
+
+database.query(`SELECT * FROM country ORDER BY SurfaceArea DESC`, (error, result) => {
```
Comment (gajduk) [inline Week1/node.js]: I believe the column name is `surface_area`

File: Week1/node.js [hunk 6]
```diff
@@ -0,0 +1,184 @@
...
+  if (error) throw error;
+
+  console.log(`\n Top 10 countries based on surface area:`);
+  result.forEach(country => {
+    console.log(country.Name);
+  });
+});
+
+// top 10 cities with the highest population?
+database.query('SELECT FROM city ORDER BY population DESC LIMIT 10', (error, result) => {
```
Comment (gajduk) [inline Week1/node.js]: select what??

File: Week1/node.js (hunk 3 above)
Comment (aalhommada) [reply Week1/node.js]: indeed i did not know , but it is implemented now , thanks a lot.

File: Week1/node.js (hunk 1 above)
Comment (aalhommada) [reply Week1/node.js]: Thanks

File: Week1/node.js (hunk 2 above)
Comment (aalhommada) [reply Week1/node.js]: FLOAT. i add it , thanks

File: Week1/node.js (hunk 3 above)
Comment (aalhommada) [reply Week1/node.js]: I did not know,  i implement it

File: Week1/node.js (hunk 4 above)
Comment (aalhommada) [reply Week1/node.js]: Thanks , i make it name

File: Week1/node.js (hunk 5 above)
Comment (aalhommada) [reply Week1/node.js]: That is right , thanks

File: Week1/node.js (hunk 6 above)
Comment (aalhommada) [reply Week1/node.js]: Thanks for mention that , it is corrected now

**Pull request closed without merging**